book/tutorials/decision_trees/01.script/01.tutorial_post_processing_xgboost_tuning.ipynb

```diff
@@ -0,0 +1,700 @@
+    "- Requires more resources: Since random forests process larger data sets, they’ll require more resources to store that data.\n",
+    "- More complex: The prediction of a single decision tree is easier to interpret when compared to a forest of them.\n",
+    "\n",
+    "#### 2.3. Extreme Gradient Boosting (XGBoost) Algorithm\n",
+    "XGBoost is one of the algorithms based on Boosting ensemble method, and the idea behind it is to train the predictors sequentially, each trying to correct its predecessor. XGBoost method tries to fit the new predictor to the residual errors made by the previous predictor. It is called gradient boosting because it uses a gradient descent algorithm to minimize the loss when adding new models. XGBoost gained significant favor in the last few years as a result of helping individuals and teams win virtually every Kaggle structured data competition. \n",
+    "\n",
+    "Advantages\n",
+    "\n",
+    "- Gradient Boosting comes with an easy to read and interpret algorithm, making most of its predictions easy to handle.\n",
+    "- Boosting is a resilient and robust method that prevents and cubs over-fitting quite easily\n",
+    "- XGBoost performs very well on medium, small, data with subgroups and structured datasets with not too many features. \n",
+    "- It is a great approach to go for because the large majority of real-world problems involve classification and regression, two tasks where XGBoost is the reigning king. \n",
+    "\n",
+    "Disadvantages \n",
+    "\n",
+    "- XGBoost does not perform so well on sparse and unstructured data.\n",
+    "- A common thing often forgotten is that Gradient Boosting is very sensitive to outliers since every classifier is forced to fix the errors in the predecessor learners. \n",
+    "- The overall method is hardly scalable. This is because the estimators base their correctness on previous predictors, hence the procedure involves a lot of struggle to streamline. \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": false,
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "source": [
+    "## 3. Setting Up the Codes and Notebook\n",
+    "\n",
+    "#### 3.1. Access the GitHub Codes\n",
+    "First we download the code files from the GitHub. In the terminal you should use the following command:"
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {
+    "collapsed": false,
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "source": [
+    "git clone https://github.com/savalann/hydromachine-tutorials"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### 3.2. Import the Python Libraries\n",
+    "Nest we will import the libraries that we need. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install hydroeval xgboost"
```

Comment:
Have you tested this on CryoCloud yet? If possible, we'd like to make sure any packages needed are pre-installed with the default environment participants will use during the hackweek. Though, I was able to run this pip install command successfully, so perhaps we don't need to pre-install these.

Reply:
Hi Steven
Thank you for the corrections. I tried it on the CryoCloud, which worked with the first pip installs. 
Do you think you need to pre-install them?
Have you run all the code?

Reply:
I can add in `py-xgboost` and `hydroeval` to CryoCloud in https://github.com/CryoInTheCloud/hub-image/pull/125 if you want?